adobe/spectrum-web-components
# fix(action-button): action-button with href can be perceived by screen reader

## Description

`<sp-button>` and `<sp-action-button>` with `href` could not be activated by mouse-click or screenreader click. 

- Updated `ButtonBase` to allow clicks when there is an `href` attribute set.  
- Added click unit tests to both `<sp-button>` and `<sp-action-button>` . 
- Added stories for `<sp-action-button>`  with an `href` attribute.

## Related issue(s)

- #4576 


## How has this been tested?

-   [ ] `<sp-button> mouse click`
    1. Go [button-accent-fill--link -  Storybook](https://nikkimk-4576-bug-action-button-href--spectrum-web-components.netlify.app/storybook/?path=/story/button-accent-fill--link)
    2. Click with a mouse.
    3. Link should be activated.
-   [ ] `<sp-button> screenreader click`
     1. Go [button-accent-fill--link - Storybook](https://nikkimk-4576-bug-action-button-href--spectrum-web-components.netlify.app/storybook/?path=/story/button-accent-fill--link)
     2. Open VoiceOver
     3. Tab focus to the button.
     4. Press <kbd>CRTL+Option+Space</kbd>  to click the link.
     5. Link should be activated.
-   [ ] `<sp-action-button> mouse click`
    1. Go [action-button--href -  Storybook](https://nikkimk-4576-bug-action-button-href--spectrum-web-components.netlify.app/storybook/?path=/story/action-button--href)
    2. Click with a mouse.
    3. Link should be activated.
-   [ ] `<sp-action-button> screenreader click`
     1. Go [action-button--href - Storybook](https://nikkimk-4576-bug-action-button-href--spectrum-web-components.netlify.app/storybook/?path=/story/action-button--href)
     2. Open VoiceOver
     3. Tab focus to the button.
     4. Press <kbd>CRTL+Option+Space</kbd>  to click the link.
     5. Link should be activated.


## Screenshots (if appropriate)

## Types of changes




## Checklist


## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
